tribe_loader.py

```diff
@@ -177,48 +192,174 @@ def load_to_tribe(main_config_file, geneset_info, prefer_update=False):
-    with open(json_filename, "w") as outfile:
-        json.dump(geneset_info, outfile, indent=2)
+def compare_genesets(tribe_genesets, processed_genesets):
+    tribe_geneset_dict = {}
+    proc_geneset_dict = {}
+
+    changed_genesets = []
+
+    # We will use the gene set 'slugs' as the identifiers for each gene set
+    for gs in tribe_genesets:
+        tribe_geneset_dict[gs['slug']] = gs
+
+    for gs in processed_genesets:
+        proc_geneset_dict[gs['slug']] = gs
+
+    changed_ctr = 0
+    unchanged_ctr = 0
+    for k, v in proc_geneset_dict.iteritems():
+        # Try to get the corresponding tribe gene set (if it exists) using
+        # the slug, which is the key in this proc_geneset_dict. Otherwise,
+        # add to changed_genesets.
+        try:
+            corr_tribe_gs = tribe_geneset_dict[k]
+        except KeyError:
+            logger.info('New gene set with slug %s is being added.',
+                        v['slug'])
+            changed_genesets.append(v)
+            changed_ctr += 1
+            continue
+
+        # Publications must be converted to a set instead of a list, because
+        # the ones in the retrieved gene sets may be the same as the ones in
+        # the processed gene sets, but just in a different order.
+        processed_annotations = {}
+        for gid, pub_list in v['annotations'].iteritems():
+            processed_annotations[gid] = set(pub_list)
+
+        retrieved_annotations = {}
+        if corr_tribe_gs['tip'] is None:
+            logger.info('Gene set with slug %s had no "tip" version. '
+                        'Adding to list of gene set versions to be created.',
+                        v['slug'])
+            changed_genesets.append(v)
+            changed_ctr += 1
+            continue
+
+        for annotation in corr_tribe_gs['tip']['annotations']:
+            gene = annotation['gene']
+            xrdb = v['xrdb']
+
+            # Get desired gene identifier
+            if xrdb == 'Entrez':
+                gene = gene['entrezid']
+            elif xrdb == 'Symbol':
+                gene = gene['systematic_name']
+            else:
+                gene = gene['xrid']
+
+            pubs = set()
+
+            for pub in annotation['pubs']:
+                pubs.add(pub['pmid'])
+
+            retrieved_annotations[gene] = pubs
+
+        if processed_annotations != retrieved_annotations:
+            logger.debug('Annotations for gene set with slug %s have changed -'
+                         ' Adding to list of gene set versions to be created.',
+                         corr_tribe_gs['slug'])
+            changed_genesets.append(v)
+            changed_ctr += 1
+
+        else:
+            unchanged_ctr += 1
+    logger.info('%s gene sets have changed, saving them to Tribe', changed_ctr)
+    logger.info('%s gene sets were unchanged', unchanged_ctr)
+    return changed_genesets
+
+
+def get_changed_genesets(species_file, secrets_file, processed_genesets,
```

Comment:
If you do rename `compare_genesets()` to `get_individual_changed_geneset()` or something like that, it probably makes sense to rename this one to `get_all_changed_genesets()`.

Reply:
Yes, this name makes more sense, I will change this function name to `get_all_changed_genesets()`